### PR TITLE
Fix bug: can not check priviate registry (not enable auth) image

### PR DESCRIPTION
**Pull request by Binyang2014**
Bug fix: image checker will failed if use private registry without authentication
issue: https://github.com/microsoft/pai/issues/4586

### CODE REVIEW

File: src/init.d/image_checker.py [hunk 1]
```diff
@@ -95,6 +96,12 @@ def __init__(self, job_config, secret):
             auth = image_info["auth"]
             self._init_auth_info(auth, secret)
 
+    def _get_registry_from_image_uri(self, image_uri) -> str:
+        if self._is_image_use_default_domain():
+            return DEFAULT_REGISTRY
+        index = self._image_uri.find("/")
+        return _get_registry_uri(image_uri[:index])
```
Comment (abuccts) [inline src/init.d/image_checker.py:103]: what if the registry uses a non-root path

File: src/init.d/image_checker.py [hunk 2]
```diff
@@ -95,6 +96,12 @@ def __init__(self, job_config, secret):
             auth = image_info["auth"]
             self._init_auth_info(auth, secret)
 
+    def _get_registry_from_image_uri(self, image_uri) -> str:
+        if self._is_image_use_default_domain():
```
Comment (abuccts) [inline src/init.d/image_checker.py:100]: "is image using" or "does image use"

File: test/test_image_checker.py [hunk 3]
```diff
@@ -216,6 +216,28 @@ def test_is_use_default_domain(self, mock):
...
                               mock_image_checker._get_normalized_image_info)
 
+    @patch.object(ImageChecker, "__init__")
+    def test_get_registry_from_image_uri(self, mock):
+        mock.return_value = None
+        mock_image_checker = ImageChecker({}, {})
+
+        test_cases = [{
+            "image_uri": "registry.domain:5000/user-name/repo-0:tag-0.version",
+            "expect_registry": "https://registry.domain:5000/v2/"
```
Comment (abuccts) [inline test/test_image_checker.py:226]: why expect https

File: src/init.d/image_checker.py [hunk 4]
```diff
@@ -86,7 +86,8 @@ def __init__(self, job_config, secret):
         image_info = docker_images[0]
 
         self._image_uri = image_info["uri"]
-        self._registry_uri = DEFAULT_REGISTRY
+        self._registry_uri = self._get_registry_from_image_uri(
```
Comment (abuccts) [inline src/init.d/image_checker.py:89]: why use https as default schema? there's no way to include schema in image uri

**Comment by Binyang2014** [comment]
> what about the fake secret bug?

After this fix, we don't need fake secret anymore. It's just a workaround

File: src/init.d/image_checker.py (hunk 4 above)
Comment (Binyang2014) [reply src/init.d/image_checker.py:89]: By default, will use `https`. If use registry use `http`, this check will be ignored.

File: src/init.d/image_checker.py (hunk 1 above)
Comment (Binyang2014) [reply src/init.d/image_checker.py:103]: If use non-root path, it will be treated use default domain

File: test/test_image_checker.py (hunk 3 above)
Comment (Binyang2014) [reply test/test_image_checker.py:226]: Will add comments for image checker. It currently only support registry which enabled `https`. For http registry, image checker will exit 0, do not block following steps

File: src/init.d/image_checker.py (hunk 1 above)
Comment (abuccts) [reply src/init.d/image_checker.py:110]: I mean the registry "domain:5000/path/"

Comment (Binyang2014) [reply src/init.d/image_checker.py:110]: Refer to https://github.com/docker/distribution/blob/742aab907b54a367e1ac7033fb9fe73b0e7344f5/reference/normalize.go#L91, seems docker only treats domain:5000 as registry path

**Pull request merged**